### ISSUE TITLE
Allow changing the game object that player controls

### ISSUE THREAD

**Issue by bitcraft** [Tuxemon/Tuxemon#509]
This must be implemented in a good way.  No hacks please.  The game should allow events to change which NPC-like object the player controls.  Currently, there are many hard-coded references to the "player object (a subclass of NPC)", but the engine  should be refactored so that the controls, camera, and menu options are not hard-coded onto the player.  The engine should allow the player NPC to be changed to another.

**Comment by Sanglorian** [comment]
Is this the tool that would allow us to temporarily assign the player a new team of tuxemon?

I'm imagining a special challenge where the player gets assigned a fixed team of tuxemon to battle with, instead of being able to bring their own. This could lead to some interesting battles:

* "Puzzle" battles where you have to figure out how to correctly use the tuxemon you are given to solve a problem. The game Neo Monsters has some great examples of this. 
* Matched battles where the two opponents have the exact same teams: it's all about how you use them.

**Comment by kerizane** [comment]
We could do it that way, or we could have a command to swap the party (and swap back later).

We might want to make other changes whilst in puzzle mode, so maybe it's best to do it your way.

**Comment by bitcraft** [comment]
> Is this the tool that would allow us to temporarily assign the player a new team of tuxemon?

Not directly related to that.  Its more like, creating a Rockitten world sprite, then allowing the player to move and control it.  Inventory, techniques, movements speed, etc will be changed to match Rockitten.

> We could do it that way, or we could have a command to swap the party (and swap back later).

Sounds like a straightforward set of game events to store, clear, and restore a NPC/Player monster list.

**Comment by Qiangong2** [comment]
> "Puzzle" battles where you have to figure out how to correctly use the tuxemon you are given to solve a problem. The game Neo Monsters has some great examples of this.

I could see this being a sort-of mini game, where you could earn items that would help you during battles.

**Comment by bitcraft** [comment]
@Qiangong2 yeah that would be interesting.  Lets keep the game mechanics on the original issue, https://github.com/Tuxemon/Tuxemon/issues/508 and talk about technical issues here.
